hub/constants.py

```diff
@@ -34,8 +35,14 @@
 DEFAULT_LOCAL_CACHE_SIZE = 0
 
 
+# meta is 100% required hub-defined meta
 DATASET_META_FILENAME = "dataset_meta.json"
 TENSOR_META_FILENAME = "tensor_meta.json"
+
+# info is 100% optional user-defined information
+DATASET_INFO_FILENAME = "dataset_info.json"
+TENSOR_INFO_FILENAME = "tensor_info.json"
```

Comment:
2 additional small files -- this means that in total we will have the following small files:

- dataset meta/dataset info (2)
- tensor info/tensor info (2 * num_tensors)
- chunk_ids (1 * num_tensors)

for a dataset with 2 tensors, there will be 8 small files.
for a dataset with 4 tensors there will be 14 small files.

the more tensors we add, the more small files they will contain. this isn't a problem now, but we should keep it in mind as we proceed.